resources/assets/lib/core/user/score-pins.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import ScoreJson from 'interfaces/score-json';
+import { route } from 'laroute';
+import { action, makeObservable, observable, runInAction } from 'mobx';
+import OsuCore from 'osu-core';
+
+export default class ScorePins {
+  @observable pins = new Map<string, boolean>();
+
+  constructor(private core: OsuCore) {
+    makeObservable(this);
+  }
+
+  apiPin(score: ScoreJson, toPin: boolean): JQuery.jqXHR<void> {
```

Comment:
return type of of the `ajax` call should be casted to be not `<any>`

Reply:
Doesn't seem to matter? 🤔

Reply:
returning `jqXHR<any>` for a function typed `jqXHR<void>` is a lint warning